src/ocean/io/serialize/StructSerializer.d

```diff
@@ -555,9 +558,20 @@ struct StructSerializer ( bool AllowUnions = false )
                 // slice array (passing a static array as ref is not allowed)
                 U[] array = field;
 
-                static if ( is(BaseTypeOfArrays!(U) == struct) )
+                alias BaseTypeOfArrays!(U) BT;
+
+                static if ( is(BT == struct) )
                 {
-                    serializeStructArray(array, field_name, serializer, data);
+                    static if ( is(BT.IsTypedef) )
```

Comment:
https://github.com/sociomantic-tsunami/ocean/blob/v3.x.x/src/ocean/meta/traits/Basic.d#L609

Reply:
I'd like to avoid an extra template instantiation when a cheap is-Expression does the trick.
also further up in this file the same is used.

Reply:
And I don't care how much memory compiler consumes if code becomes more readable.

We generates thousands of template instances with versioned serializer, don't do micro-optimizations in code that doesn't matter.

Reply:
> also further up in this file the same is used

Please adjust it too in that case.

Reply:
As you wish :)

Reply:
./src/ocean/io/serialize/StructSerializer.d(590): Error: ocean.core.Traits.isTypedef(T) at ./src/ocean/core/Traits.d(873) conflicts with ocean.meta.traits.Basic.isTypedef(T) at ./src/ocean/meta/traits/Basic.d(609)